lib/Models/SearchProviders/CatalogIndex.ts

```diff
@@ -31,10 +31,20 @@ export default class CatalogIndex {
     return this._searchIndex;
   }
 
-  readonly loadPromise: Promise<void>;
+  private _loadPromise: Promise<void> | undefined;
```

Comment:
Should this be an `observable`? Considering we use it in a computed here:

https://github.com/TerriaJS/terriajs/blob/7510a87c0041f0ac4d2a9fe98dc4fc7b395c5ae5/lib/Models/SearchProviders/CatalogSearchProvider.ts#L118-L123

Reply:
Oh yeah good pickup!